src/main/java/seedu/module/model/module/Module.java

```diff
@@ -10,4 +10,6 @@
     String getTitle();
 
     String getDescription();
+
+    Deadline getDeadline();
```

Comment:
In my opinion, `ArchivedModule` shouldn't have a deadline field. So this interface method does not make sense. I would like to suggest either we

1) Use a default method here (which I'm not entirely comfortable with), or

2) Create a new interface that contains methods only a `TrackedModule` has, so that we can also use that interface to display information only `TrackedModule` has